### DISCUSSION TITLE
What would be the best way to customize / or use our own License File, Dialog Banner and Dialog Image in ManagedUI?

### DISCUSSION THREAD

**Discussion by se** [oleg-shilo/wixsharp#1397] [General]
In ManagedUI class, I see the following lines adding Binary files to the project.

![image](https://github.com/oleg-shilo/wixsharp/assets/491844/b97d6700-45d9-4b11-80eb-1c11bfbfb11a)

I have tried to customize that but couldn't able to do it. Is it possible to check current binary files and add them if they are not exists @oleg-shilo ?

Like this;

```csharp
public void BeforeBuild(ManagedProject project)
{
    var file = LocalizationFileFor(project);
    ValidateUITextFile(file);
    project.AddBinary(new Binary(new Id("WixSharp_UIText"), file));

    if (!project.Binaries.Any(x => x.Id == "WixSharp_LicenceFile"))
    {
        project.AddBinary(new Binary(new Id("WixSharp_LicenceFile"), LicenceFileFor(project)));
    }
    if (!project.Binaries.Any(x => x.Id == "WixUI_Bmp_Dialog"))
    {
        project.AddBinary(new Binary(new Id("WixUI_Bmp_Dialog"), DialogBitmapFileFor(project)));
    }
    if (!project.Binaries.Any(x => x.Id == "WixUI_Bmp_Banner"))
    {
        project.AddBinary(new Binary(new Id("WixUI_Bmp_Banner"), DialogBannerFileFor(project)));
    }
}
```

**Comment by se** [comment]
OK, this is my bad.

I could able to change ManagedUI's images like following;

```csharp
project.LicenceFile = @"Resources\LICENSE.rtf";
project.BackgroundImage = @"Resources\installer-dialog.png";
project.BannerImage = @"Resources\installer-banner.png";
```

**Reply by juagicre** [reply]
This works for me in case of the LicenceFile but it does not in case of the BannerImage, even setting it directly in the xaml in the WelcomeDialog.xaml it does not show up -_-

**Comment by oleg-shilo** [comment]
If you are using WixSharp stock WPF dialogs then it seems to work. I have attached the code sample.

However, if you are using WiX3 version of the WixSharp VS template for "WPF Custom UI" then the template has a mistake.

```C#
// wrong
public BitmapImage Banner => session?.GetResourceBitmap("WixSharpUI_Bmp_Banner").ToImageSource();

// right
public BitmapImage Banner => session?.GetResourceBitmap("WixUI_Bmp_Banner").ToImageSource();

```

This mistake came from the WiX stream accidentally.

In the attached sample I have also corrected it.


[WixSharp_Setup63.zip](https://github.com/oleg-shilo/wixsharp/files/13715003/WixSharp_Setup63.zip)

**Reply by juagicre** [reply]
Thanks for the reply, oleg!
I did find the way to bind even more images and add them as bitmap resources by try and error, but thanks for the clarification here!
